### PR TITLE
doc: add README.md

### PR DESCRIPTION
**Pull request by volodeyka**
This change is [Reviewable](https://reviewable.io/reviews/volodeyka/event-struct/30)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,14 +4,38 @@
...
-convallis velit sed, accumsan leo. Fusce sed dui est. Nullam suscipit tincidunt
-ipsum id accumsan. Sed eget sapien ornare magna fermentum blandit. Maecenas
-commodo porttitor leo, at pellentesque nulla imperdiet vel. Ut iaculis, lectus
-eget sodales bibendum, justo nulla posuere augue, eu rutrum turpis dui ac erat.
-Donec pharetra elit vitae tincidunt vestibulum. Sed iaculis facilisis ipsum sed
-aliquam. Nullam tincidunt purus sed est pulvinar lacinia. Integer at nunc vitae
-sem pulvinar maximus quis in ipsum. Duis ut sem varius, ultrices leo vel,
-interdum orci. Etiam bibendum varius lectus ac scelerisque. Nam in magna sapien.
-Nunc mattis orci non risus rutrum posuere. Mauris nec viverra lectus.
+A library of formalized event structure theory applied to weak-memory models. Includes notion and theory of execution event structures, labeled transition system on execution event structures, big-step semantics for simple register machine, and well-founded relations theory.
```
Comment (anton-trunov) [inline README.md:7]: There are two usual conventions for formatting:
- 80-character text width (text editors support this automatically or have a key binding to reformat quickly)
- one sentence per line, this makes diffs nicer

**Review by anton-trunov — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -4,14 +4,44 @@
...
-convallis velit sed, accumsan leo. Fusce sed dui est. Nullam suscipit tincidunt
-ipsum id accumsan. Sed eget sapien ornare magna fermentum blandit. Maecenas
-commodo porttitor leo, at pellentesque nulla imperdiet vel. Ut iaculis, lectus
-eget sodales bibendum, justo nulla posuere augue, eu rutrum turpis dui ac erat.
-Donec pharetra elit vitae tincidunt vestibulum. Sed iaculis facilisis ipsum sed
-aliquam. Nullam tincidunt purus sed est pulvinar lacinia. Integer at nunc vitae
-sem pulvinar maximus quis in ipsum. Duis ut sem varius, ultrices leo vel,
-interdum orci. Etiam bibendum varius lectus ac scelerisque. Nam in magna sapien.
-Nunc mattis orci non risus rutrum posuere. Mauris nec viverra lectus.
+A library of formalized event structure theory applied to weak-memory models.
```
Comment (eupp) [inline README.md:7]: I would propose to rename the title of the repo into `Mechanized Theory of Event Structures`. 
Then, I would rephrase this sentence as `A Coq library of formalized event structure theory with applications to concurrency semantics` (I would not fix `memory models` here, because in the future we can also look at other domains and applications).

File: README.md [hunk 3]
```diff
@@ -4,14 +4,44 @@
...
+Includes notion and theory of execution event structures, labeled transition
+system on execution event structures, big-step semantics for simple
+register machine, and well-founded relations theory.
```
Comment (eupp) [inline README.md:10]: I think this is too many details for the abstract. You just enlist all the things we have implemented so far. Are you going to rewrite the abstract every time we add a new feature?)

I would rather remove it, or try to come up with a more general description of the library.

File: README.md [hunk 4]
```diff
@@ -4,14 +4,44 @@
...
 
+## Files in poject
```
Comment (eupp) [inline README.md:12]: It is a good idea to give a quick description of each file  :+1: 

I would only change the header to `Description of Files`

File: README.md [hunk 5]
```diff
@@ -4,14 +4,44 @@
...
+All files include description of its constituents.
```
Comment (eupp) [inline README.md:13]: `constituents` --> `content`

File: README.md [hunk 6]
```diff
@@ -4,14 +4,44 @@
...
+
+### `utilities.v`
+Some extra lemmas and tactics to make our work more convenient.
+Includes `lia` analouge for ssreflect -- `slia`.
+Based on: https://github.com/amahboubi/lia4mathcomp
+
+### `inhtype.v`
+Interface for inhabited type
```
Comment (eupp) [inline README.md:21]: Could you please add dot `.` at the end of sentences

File: README.md [hunk 7]
```diff
@@ -4,14 +4,44 @@
...
+
+### `wftype.v`
+Interface for types with well-founded order
+
+### `ident.v`
+Interface for types that can be used as identifiers.
+In other words we, require:
+1) `ident0` -- first identifier
+2) `<` -- well-founded order
+3) `fresh : T -> T` -- function that returns fresh identifier
```
Comment (eupp) [inline README.md:31]: Maybe add an axiom on `fresh` here?

File: README.md [hunk 8]
```diff
@@ -4,14 +4,44 @@
...
+
+### `relations.v`
+Theory of well-founded orders closures
```
Comment (eupp) [inline README.md:34]: `Theory of transitive closure of well-founded relations`

File: README.md [hunk 9]
```diff
@@ -4,14 +4,44 @@
...
+on whole type `E` and embedded in fixed relation. 
+
+### `eventstructure.v`
+Theory of finite execution event structures
+
+### `transitionsystem.v`
+Labeled transition system on execution event structures
+
+### `regmachine.v`
+Big-step semantics for simple register machine
```
Comment (eupp) [inline README.md:47]: It's actually a `small-step` semantics :)

**Review by eupp — CHANGES_REQUESTED** [review]

File: README.md (hunk 3 above)
Comment (volodeyka) [reply README.md:10]: I just tried to follow some README templates in Coq-community
Actually, I agree with you. What about: 
"This library was dedicated to the formalization of some weak memory models in terms of event structures. But also includes the theory of well-founded relations"

Comment (anton-trunov) [reply README.md:10]: > I just tried to follow some README templates in Coq-community

Which specific description did you follow? This sounds like a bug that should be fixed.

File: README.md [hunk 10]
```diff
@@ -1,47 +1,48 @@
...
 
-A library of formalized event structure theory applied to weak-memory models.
+A Coq library of formalized event structure theory with applications to concurrency semantics.
 Includes notion and theory of execution event structures, labeled transition
 system on execution event structures, big-step semantics for simple
 register machine, and well-founded relations theory.
 
-## Files in poject
-All files include description of its constituents.
+## Description of Files
```
Comment (anton-trunov) [inline README.md:12]: Grammar nits: this section lacks a number of articles ("a" and/or "the" mostly). While it's fine to skip them in section titles but I'd prefer having them in the text body.

File: relations.v [hunk 11]
```diff
@@ -80,29 +81,27 @@ Proof.
   exact: filter_lt.
 Qed. 
 
+Print  t1n_trans.
```
Comment (anton-trunov) [inline relations.v:84]:
```suggestion
```

Comment (anton-trunov) [reply relations.v:84]: Wait, why is this code change in a documentation PR?

File: README.md (hunk 10 above)
Comment (volodeyka) [reply README.md:12]: Oh, I have some problems with them, but I'll try to fix it

Comment (anton-trunov) [reply README.md:12]: Oh, ok, I thought that was a stylistic choice :)
